### PR TITLE
Adds support for Polygon, Binance, and Avalanche networks

### LINKED ISSUE #3: Allow user-specified block explorer
Similar to how we added Polygonscan/Bscscan options in the `lattice-manager`, we should allow the user to specify that here. Currently all addresses are looked up using Etherscan, but we should add an attribute to metadata called `scannerUrl`, which can be e.g. `https://polygonscan.com`.

The `lattice-manager` should also have a dropdown to switch between networks to look for packs associated with that network. For example, the default should only load packs with either `etherscan.io` or `null` `scannerUrl`.

### PR DESCRIPTION
**Pull request by douglance**
Fixes #3

### CODE REVIEW

File: script.ts [hunk 1]
```diff
@@ -18,34 +18,81 @@ To run this script, you will need a working Etherscan API key:
...
 const superagent = require("superagent");
 const parseAbi = new SDK({ crypto: require("crypto") }).parseAbi;
 const jsonc = require("jsonc");
 const Throttle = require("superagent-throttle");
-
+const DEFAULT_NETWORK = "ethereum";
+const CONTRACT_NETWORKS = {
+  ethereum: {
+    baseUrl: "https://api.etherscan.io",
+    apiKey: process.env.ETHERSCAN_KEY,
```
Comment (alex-miller-0) [inline script.ts:30]: Just curious... where do these api keys get defined in CI?

File: script.ts [hunk 2]
```diff
@@ -18,34 +18,81 @@ To run this script, you will need a working Etherscan API key:
...
+    apiRoute: "api?module=contract&action=getabi&address=",
+    throttle: new Throttle({
+      active: true,
+      rate: 5,
+      ratePer: 1000,
+      concurrent: 1,
+    }),
```
Comment (alex-miller-0) [inline script.ts]: You could save this object as a constant to avoid the copypasta

File: types/index.ts [hunk 3]
```diff
@@ -12,7 +15,22 @@ export interface Contract {
   addresses: TaggedAddress[];
 }
 
+export interface ProcessedContract extends Contract {
+  network: string;
```
Comment (alex-miller-0) [inline types/index.ts]: Should this be a `Network` rather than a `string`?

File: script.ts (hunk 1 above)
Comment (douglance) [reply script.ts:30]: I meant to mention that. We add them in the settings of the repo itself. I put in the Etherscan key that I generated, but we might want to update that to an official one, and we'll need to add keys for the other services.

File: script.ts (hunk 2 above)
Comment (douglance) [reply script.ts]: Good point 🤦

File: types/index.ts (hunk 3 above)
Comment (douglance) [reply types/index.ts]: Nice catch. Thanks!

**Comment by alex-miller-0** [comment]
@douglance pls post commit hashes against the comments so I know when to resolve them

File: types/index.ts (hunk 3 above)
Comment (douglance) [reply types/index.ts]: `8289219524253097fac4247b40c3069ed06fb569`

File: script.ts (hunk 2 above)
Comment (douglance) [reply script.ts]: `8289219524253097fac4247b40c3069ed06fb569`

**Review by alex-miller-0 — APPROVED** [review]

**Pull request merged**